logstash-core/spec/logstash/agent_spec.rb

```diff
@@ -425,12 +473,13 @@ class DummyOutput2 < LogStash::Outputs::DroppingDummyOutput; end
       Thread.abort_on_exception = true
 
       @t = Thread.new do
+        Thread.current.abort_on_exception = true
```

Comment:
Wait, this is a thing?

BTW, if you want a thread you can capture the exception from, you can use Stud::Task which basically wraps Thread and lets you get the exception - http://www.rubydoc.info/gems/stud/Stud/Task

```ruby
@t = Stud::Task.new do ... end
...
result, exception = @t.wait
```